python/pyiceberg/schema.py

```diff
@@ -1364,3 +1364,11 @@ def _(file_type: DecimalType, read_type: IcebergType) -> IcebergType:
             raise ResolveError(f"Cannot reduce precision from {file_type} to {read_type}")
     else:
         raise ResolveError(f"Cannot promote an decimal to {read_type}")
+
+
+@promote.register(FixedType)
+def _(file_type: FixedType, read_type: IcebergType) -> IcebergType:
+    if isinstance(read_type, UUIDType) and len(file_type) == 16:
```

Comment:
One minor request. Could you add a small comment here explaining why this promotion is there? That PyArrow reads a Parquet UUID as FixedBytes, and therefore we need to allow this promotion.

Reply:
Added a comment😊

Reply:
Thanks!